src/gl/texture.c

```diff
@@ -1647,7 +1647,7 @@ void gl4es_glTexSubImage2D(GLenum target, GLint level, GLint xoffset, GLint yoff
         pixel_to_ppm(pixels, width, height, format, type, bound->texture, glstate->texture.pack_align);
     }
 
-    int callgeneratemipmap = 0;
+    uint8_t callgeneratemipmap = 0;
     if ((target!=GL_TEXTURE_RECTANGLE_ARB) && (bound->mipmap_need || bound->mipmap_auto)) {
```

Comment:
why the `uint8_t`? what's wrong with a simple `int`?

Reply:
Since you're using it as a simple boolean expression, using int (4 bytes) just uses more memory than necessary. Now it only uses 1 byte :)

Reply:
I think access to int may be faster than access to byte. But I'm unsure. 

Anyway, it's variable allocated on the stack, so each one will be aligned the size of (void*) anyway, there is absolutly no change in memory consumed here.

Reply:
Hmm yep, I did not think about that. In this case, since this variable only exists during the function lifetime, it'll be in the stack.

I think you can edit the commit to remove those changes then, I've allowed edits by maintainers.